### PR TITLE
[CIRCLE-32867] Adding contexts field restrictions doc

### PR DESCRIPTION
**Pull request by roncodingenthusiast**
# Description
Adding context fields validation/restrictions docs

# Reasons
[CIRCLE-32867](https://circleci.atlassian.net/browse/CIRCLE-32867)

* Add criteria for validation for environment variable names/values
* Add docs on maximum number of environment variable per contexts
* Add docs on maximum number of context per organizations
* Add criteria for validation for context names

[CIRCLE-32867]: https://circleci.atlassian.net/browse/CIRCLE-32867

### CODE REVIEW

File: jekyll/_cci2/contexts.md [hunk 1]
```diff
@@ -254,6 +264,21 @@ endpoint with the appropriate request body. For this request, replace the
 environment variable name. The request body should include a `value` key 
 containing the plaintext secret as a string.
 
+##### Environment Variable validation
+
+###### Environment variable names
+
+The environment variable name must be 300 or fewer characters, must have alpha or _ as the first character and can contain alphanumeric or _ as the remaining characters.
```
Comment (ellipses) [inline jekyll/_cci2/contexts.md:271]: and can *only* contain

File: jekyll/_cci2/contexts.md [hunk 2]
```diff
@@ -254,6 +264,21 @@ endpoint with the appropriate request body. For this request, replace the
 environment variable name. The request body should include a `value` key 
 containing the plaintext secret as a string.
 
+##### Environment Variable validation
```
Comment (ellipses) [inline jekyll/_cci2/contexts.md:267]: 3 `#`s right? So it's not included as part of `Secure Environment Variable Creation, Deletion, and Rotation`

File: jekyll/_cci2/contexts.md (hunk 1 above)
Comment (teesloane) [reply jekyll/_cci2/contexts.md:271]: Can we also wrap underscores in in `backticks` for easier reading?

File: jekyll/_cci2/contexts.md (hunk 2 above)
Comment (roncodingenthusiast) [reply jekyll/_cci2/contexts.md:267]: this is 5 `#`s because it falls under `#### Creating Environment Variables`

File: jekyll/_cci2/contexts.md [hunk 3]
```diff
@@ -254,6 +264,21 @@ endpoint with the appropriate request body. For this request, replace the
...
+
+###### Environment variable names
+
+The environment variable name must be 300 or fewer characters, must have alpha or `_` as the first character and can *only* contain  alphanumeric or `_` as the remaining characters.
+
+###### Environment variable values
+
+An environment variable value must be 32000 or fewer characters and must NOT contain null.
```
Comment (teesloane) [inline jekyll/_cci2/contexts.md:275]:
```suggestion
An environment variable value must be 32000 or fewer characters and must NOT contain `null`. 
```

File: jekyll/_cci2/contexts.md [hunk 4]
```diff
@@ -254,6 +264,21 @@ endpoint with the appropriate request body. For this request, replace the
...
+##### Environment Variable validation
+
+###### Environment variable names
+
+The environment variable name must be 300 or fewer characters, must have alpha or `_` as the first character and can *only* contain  alphanumeric or `_` as the remaining characters.
+
+###### Environment variable values
+
+An environment variable value must be 32000 or fewer characters and must NOT contain null. 
+**Note**: Empty is a valid option for environment variable values.
```
Comment (teesloane) [inline jekyll/_cci2/contexts.md:276]:
```suggestion
**Note**:  An empty environment variable is  considered valid.
```

**Review by teesloane — CHANGES_REQUESTED** [review]

**Review by teesloane — APPROVED** [review]

File: jekyll/_cci2/contexts.md [hunk 5]
```diff
@@ -117,6 +117,17 @@ jobs:
...
 If you move your repository to a new organization, you must also have the context with that unique name set in the new organization.
 
+### Contexts & Environment variables validation
+
+When creating contexts/environment variables, please note the following:
+
+- The context name must be 200 or fewer characters, must contain at least one non-whitespace character, and must not contain leading, trailing or vertical whitespaces.
+- An organization may contain no more than 500 contexts.
+- The environment variable name must be 300 or fewer characters, must have alpha or `_` as the first character and can *only* contain  alphanumeric or `_` as the remaining characters.
+- An environment variable value must be 32000 or fewer characters and must NOT contain `null`.
```
Comment (zalimeni) [inline jekyll/_cci2/contexts.md:127]: I think we can leave out the bit about `NUL` byte - it's not doc'd by any of our competitors, but it's invalid for any POSIX environment and I don't know of a significant use case for it.
```suggestion
- An environment variable value must have 32K or fewer characters. 
```

File: jekyll/_cci2/contexts.md [hunk 6]
```diff
@@ -117,6 +117,17 @@ jobs:
...
+### Contexts & Environment variables validation
+
+When creating contexts/environment variables, please note the following:
+
+- The context name must be 200 or fewer characters, must contain at least one non-whitespace character, and must not contain leading, trailing or vertical whitespaces.
+- An organization may contain no more than 500 contexts.
+- The environment variable name must be 300 or fewer characters, must have alpha or `_` as the first character and can *only* contain  alphanumeric or `_` as the remaining characters.
+- An environment variable value must be 32000 or fewer characters and must NOT contain `null`. 
+**Note**: An empty environment variable is considered valid.
+- A context may contain no more than 100 environment variables.
```
Comment (zalimeni) [inline jekyll/_cci2/contexts.md:129]: ~~ Same here, just an alternate wording that I thought might work - totally happy either way.
```suggestion
- Each context is limited to 100 environment variables.
```

**Comment by zalimeni** [comment]
@roncodingenthusiast sorry, this is what I get for leaving up a stale PR window 😅  if you like any of the above suggestions, feel free to crib and I'll rubber-stamp, but don't block on me.

**Review by amilcaru123 — COMMENTED** [review]
[]()

**Pull request merged**